src/QsCompiler/DataStructures/SyntaxTree.fs

```diff
@@ -789,11 +791,42 @@ type QsNamespace = {
     member this.WithElements (getElements : Func<_,_>) = {this with Elements = getElements.Invoke(this.Elements)}
 
 
+type TypeParameterName = QsQualifiedName * NonNullable<string>
+type TypeParameterResolutions = ImmutableDictionary<TypeParameterName, ResolvedType>
+
+// Interface used to represent an edge in the call graph
+type ICallGraphEdge =
+    inherit  IEquatable<ICallGraphEdge>
+    abstract member ParamResolutions : TypeParameterResolutions
+    abstract member Parent : QsQualifiedName
+    abstract member ReferenceRange : DataTypes.Range
+
+// Interface used to represent a node in the call graph
+type ICallGraphNode =
+    inherit IEquatable<ICallGraphNode>
+    abstract member CallableName : QsQualifiedName
+    abstract member Kind : QsSpecializationKind
+    abstract member TypeArgs : QsNullable<ImmutableArray<ResolvedType>>
+
+// Interface used to represent a call graph
+type ICallGraph = 
+    abstract member Count : int
+    abstract member Nodes : ImmutableHashSet<ICallGraphNode>
+    abstract member ContainsNode : ICallGraphNode -> bool
+    abstract member GetAllDependencies : ICallGraphNode -> ILookup<ICallGraphNode, ICallGraphEdge>
+    abstract member GetAllDependencies : QsSpecialization -> ILookup<ICallGraphNode, ICallGraphEdge>
+    abstract member GetDirectDependencies : ICallGraphNode -> ILookup<ICallGraphNode, ICallGraphEdge>
+    abstract member GetDirectDependencies : QsSpecialization -> ILookup<ICallGraphNode, ICallGraphEdge>
+    abstract member VerifyAllCycles : Unit -> IEnumerable<QsQualifiedName * QsCompilerDiagnostic>
```

Comment:
Can you move the documentation comments from the call graph to here?

Reply:
I've copied over the docs as you suggested.